[flutter_releases] Flutter stable 3.24.0 Framework Cherrypicks (#154430)
flutter/flutter: CHANGELOG.md

-->
## Flutter 3.24 Changes

> 

Reviewer: Were there any Dart changes included in this hotfix?

Author: https://github.com/dart-lang/sdk/blob/stable/CHANGELOG.md#352---2024-08-28

Author: Dart patch notes updated to the format Dart uses.

### [3.24.2](https://github.com/flutter/flutter/releases/tag/3.24.2)

> - [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fix a crash when deleting text inside CupertinoPageRoute, with a CJK (chinese, japanese, korean) keyboard on Android.

Reviewer: ```suggestion
- [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fixes a crash on Android when deleting `EditableText` inside `CupertinoPageRoute`, with a CJK (chinese, japanese, korean) keyboard.
```

> - [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue where Flutter TextField may stop accepting input on iOS.

Reviewer: ```suggestion
- [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue on iOS where Flutter `TextField`'s may stop accepting input.
```

> - [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank when a SelectionArea/SelectableRegion is used as a child of a Scrollable like ListView or PageView on Android and iOS.

Reviewer: ```suggestion
- [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank on Android and iOS when a `SelectionArea`/`SelectableRegion` is used as a child of a Scrollable like `ListView` or `PageView`.
```

> - [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.

Reviewer: I don't understand where this comes in.  Is this when we pass a specific flag.  Why specifically Android?

> - [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.

Reviewer: Thanks for clarifying.

> - [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.

Author: Going with "Excessive logging" (thanks go/duckie)

> - [flutter/153967](https://github.com/flutter/flutter/pull/153967): Fixes macOS host build failure when there are no native asset frameworks to codesign. Impacts users who have enabled experimental native assets feature.

Reviewer: ```suggestion
- [flutter/153967](https://github.com/flutter/flutter/pull/153967): Fixes a host build failure on macOS when the `native assets` experiment is enabled, and there are no native asset frameworks to codesign.  
```

> - [flutter/153769](https://github.com/flutter/flutter/pull/153769): When running a Flutter app, display a concise error message when connection to the device is lost.

Reviewer: ```suggestion
- [flutter/153769](https://github.com/flutter/flutter/pull/153769): Simplifies the error message when the connection is lost with a connected device when running a Flutter app.
```

> - [flutter/154270](https://github.com/flutter/flutter/pull/154270): Prevent preemptive gradle crash for android builds that would fail to build anyway but with a confusing error message.

Reviewer: As a user, this message seems confusing.

> - [flutter/54735](https://github.com/flutter/engine/pull/54735): When semantics are enabled on Web, the onTap of various widgets (GestureDetector, InkWell) are called twice or the callback from one widget is called when another widget is pressed.

Reviewer: ```suggestion
- [flutter/54735](https://github.com/flutter/engine/pull/54735):  Fixes an error on Flutter Web where `onTap` is called twice on various widgets (`GestureDetector`, `InkWell`) when semantics are enabled.
```

-->
## Flutter 3.24 Changes

### [3.24.2](https://github.com/flutter/flutter/releases/tag/3.24.2)
- Fixes a bug where `ZLibDecoder` would incorrectly attempt to decompress data
  past the end of the zlib footer (issue [#56481][]).
- Fixes issue where running `dart` from `PATH` could result in some commands not
  working as expected (issues [#56080][], [#56306][], [#56499][]).
- Fixes analysis server plugins not receiving `setContextRoots` requests or
  being provided incorrect context roots in multi-package workspaces (issue
  [#56475][]).
- [flutter/153949](https://github.com/flutter/flutter/issues/153949): Fixes a crash on Android when deleting `EditableText` inside `CupertinoPageRoute`, with a CJK (chinese, japanese, korean) keyboard.
- [flutter/153939](https://github.com/flutter/flutter/issues/153939): Fixes an issue on iOS where Flutter `TextField`s may stop accepting input.
- [flutter/152420](https://github.com/flutter/flutter/issues/152420): Fixes scrolling jank on Android and iOS when a `SelectionArea`/`SelectableRegion` is used as a child of a Scrollable like `ListView` or `PageView`.

> - [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes log spam when building a freshly created template app for Android.

Author: ```suggestion
- [flutter/154199](https://github.com/flutter/flutter/pull/154199): Removes excessive logging when building a freshly created template app for Android.
```
